lib/fs.js

```diff
@@ -3102,6 +3104,37 @@ function createWriteStream(path, options) {
   return new WriteStream(path, options);
 }
 
+const lazyGlob = getLazy(() => require('internal/fs/glob').Glob);
+
+function glob(pattern, options, callback) {
+  emitExperimentalWarning('glob');
+  if (typeof options === 'function') {
+    callback = options;
+    options = undefined;
+  }
+  callback = makeCallback(callback);
+
+  const Glob = lazyGlob();
+  (async () => {
+    try {
+      const res = [];
+      for await (const entry of new Glob(pattern, options).glob()) {
+        res.push(entry);
```

Comment:
We should maybe add a todo to switch to using the iterator helper proposal when that will have landed
```suggestion
        ArrayPrototypePush(res, entry);
```